### PR TITLE
Complete corrections for multiple-determination reasons.

### PR DESCRIPTION
**Pull request by TreyE**
# PR Checklist

Please check if your PR fulfills the following requirements
- [X] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [X] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [X] For all UI changes, there is cucumber coverage
- [X] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [X] Any endpoint modified in the PR only responds to the expected MIME types.
- [X] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [X] There are no inline styles added
- [X] There are no inline javascript added
- [X] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [X] Code does not use .html_safe
- [X] All images added/updated have alt text
- [X] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/188151062

# A brief description of the changes

Current behavior: Assistance applications sent to medicaid gateway do not support multiple reasons and sometimes default to being specified as a 'FullDetermination' when they should not be.

New behavior:  Assistance applications now allow for multiple reasons, and the logic for which primary reason to set has been corrected.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name: `MULTIPLE_DETERMINATION_SUBMISSION_REASONS_IS_ENABLED`

### CODE REVIEW

File: components/financial_assistance/app/domain/financial_assistance/operations/applications/transformers/application_to/cv3_application.rb [hunk 1]
```diff
@@ -248,17 +248,30 @@ def applicants(application, benchmark_premiums)
...
               end
             end
-                        # rubocop:enable Metrics/AbcSize
+            # rubocop:enable Metrics/AbcSize
             # rubocop:enable Metrics/MethodLength
 
+            def select_primary_reason_code(applicant, application)
+              if applicant.is_gap_filling
+                "GapFilling"
+              elsif application.request_full_determination
```
[the remark this answers is not in the document]
Comment (mdkaraman) [reply components/financial_assistance/app/domain/financial_assistance/operations/applications/transformers/application_to/cv3_application.rb:266]: @saipraveen18 good catch, seems like this field isn't in use anymore (or possibly was never used).

[the remark this answers is not in the document]
Comment (TreyE) [reply components/financial_assistance/app/domain/financial_assistance/operations/applications/transformers/application_to/cv3_application.rb:266]: I'll fix it to use the correct attribute then, thanks!

**Review by saipraveen18 — APPROVED** [review]
LGMT 👍

**Review by mdkaraman — APPROVED** [review]
Looks good!

**Pull request merged**